### PR TITLE
Weekly Sync PR

### PR DESCRIPTION
**Pull request by abhi18av**
The preview is deployed here https://nf-tower-docs-staging-html.netlify.app/22.1/

### CODE REVIEW

File: docs/compute-envs/azure-batch.md [hunk 1]
```diff
@@ -6,7 +6,11 @@ description: 'Step-by-step instructions to set up Azure Batch in Nextflow Tower.
 ## Overview
 
 !!! warning
-    The Tower support for Azure Batch is currently in beta. Any feedback and suggestions are welcome.    
+    The Tower support for Azure Batch is currently in beta. Any feedback and suggestions are welcome.
+
+    In order to manage capacity during the global health pandemic, Microsoft has reduced core quotas for new Batch accounts. Depending on your region and subscription type, a newly-created account may not be entitled to any VMs without first making a service request to Azure. 
+
+    Please see Azure's [Batch service quotas and limits](https://docs.microsoft.com/en-us/azure/batch/batch-quota-limit#view-batch-quotas) page for further details.
```
Comment (abhi18av) [inline docs/compute-envs/azure-batch.md:13]:
```suggestion
    Depending on your region and subscription type, a newly-created account may not be entitled to any VMs without first making a service request to Azure. Please see Azure's [Batch service quotas and limits](https://docs.microsoft.com/en-us/azure/batch/batch-quota-limit#view-batch-quotas) page for further details.
```



Shrinking the text a bit since the rendering didn't look nice on the preview website. 

image

File: docs/faq.md [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+Please verify the following:
+
+1. The `MICRONAUT_ENVIRONMENTS` environment variable associated with the `backend` container:
+    * Contains `prod,redis,ha`
+    * Does not contain `cron`
+2. The `MICRONAUT_ENVIRONMENTS` environment variable associated with the `cron` container:
+    * Contains `prod,redis,cron`
+    * Does not contain `ha`
+3. You do not have another copy of the `MICRONAUT_ENVIRONMENTS` environment variable defined elsewhere in your application (e.g. a _tower.env_ file or Kubernetes ConfigMap).
+4. If you are using a separate container/pod to execute _migrate-db.sh_, there is no `MICRONAUT_ENVIRONMENTS` environment variable assigned to it.
```
Comment (abhi18av) [inline docs/faq.md:50]: @gwright99 , I'm thinking that we should remove references to the information which only an enterprise user would know.

If you agree I can port them over to the Deployment docs - what do you think?

File: docs/faq.md [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+
+
+**<p data-question>Q: "No such variable" error.**</p>
+
+This error can occur if you execute a DSL 1-based Nextflow workflow using [Nextflow 22.03.0-edge](https://github.com/nextflow-io/nextflow/releases/tag/v22.03.0-edge) or later.
```
Comment (abhi18av) [inline docs/faq.md:55]: This is a perfect example of the crucial information people might expect from Tower Usage docs 👍

File: docs/faq.md [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+
+### Compute Environments
+
+**<p data-question>Q: Can the name of a Compute Environment created in Tower contain special characters?**
+
+No. Tower version 21.12 and later do not support the inclusion of special characters in the name of Compute Environment objects.
+
+
+### Configuration
+
```
Comment (abhi18av) [inline docs/faq.md:66]: This section is relevant for Tower instance admins therefore the deployment docs might be better for this?

File: docs/faq.md [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+**<p data-question>Q: Can a custom path be specified for the `tower.yml` configuration file?**</p>
+
+Yes. Provide a POSIX-compliant path to the `TOWER_CONFIG_FILE` environment variable.
+
+
+**<p data-question>Q: Why do parts of `tower.yml` not seem to work when I run my Tower implementation?**</p>
+
```
Comment (abhi18av) [inline docs/faq.md:73]: The `tower.yml` can now be used for two (distinct) purposes

1. Customizing the Enterprise Tower installation
2. Providing the path to reports which are generated via a Nextflow pipeline

File: docs/faq.md [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+tower:
+  pipeline:
+    allow-local-repos:
+      - /path/to/repo
+```
+
+Note: This feature is not available to Tower Cloud users.
+
+
+### Logging
```
Comment (abhi18av) [inline docs/faq.md:127]: This section is also relevant to Tower admins.

File: docs/faq.md [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+As part of the AWS Batch creation process, Tower Forge will set ECS Agent parameters in the EC2 Launch Template that is created for your cluster's EC2 instances:
+
+* For clients using Tower Enterprise v22.01 or later:
+    * Any AWS Batch environment created by Tower Forge will set the ECS Agent's `ECS_IMAGE_PULL_BEHAVIOUR` set to `once`. 
+* For clients using Tower Enterprise v21.12 or earlier:
+    * Any AWS Batch environment created by Tower Forge will set the ECS Agent's `ECS_IMAGE_PULL_BEHAVIOUR` set to `default`.
+
+Please see the [AWS ECS documentation](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-agent-config.html) for an in-depth explanation of this difference. 
+
+**</p>Note:**</p> This behaviour cannot be changed within the Tower Application.
```
Comment (abhi18av) [inline docs/faq.md:211]: @gwright99 , this does make sense for end users but I'm not sure if we should refer Enterprise version `22.01` or `21.12` in these docs?

The usage docs follows the versions on tower.nf rather than the enterprise releases.

**Review by abhi18av — COMMENTED** [review]
I think this PR is in good shape now and we can merge it once the scope of some FAQ sections ( Usage/Installation docs) is finalized.

File: docs/faq.md (hunk 3 above)
Comment (abhi18av) [reply docs/faq.md:55]: Just finished the daily call - actually, this error message might change within NF.

The default on Tower is still meant to be DSL1 so perhaps we can delegate this to the NF layer?

File: docs/faq.md (hunk 2 above)
Comment (gwright99) [reply docs/faq.md:50]: I think it does not make sense to have Deployment and Help living in their own websites, and we have to choose what goes where. Given that all this material is now public, I think we should be looking to merge the two content bases into one unified body.

This position is based on my assumption that an Enterprise client will be using the help.tower.nf site to figure out day-to-day activities and troubleshooting. The error itself will occur as a result of someone trying to run a pipeline in Tower; it just happens to originate in an incorrect configuration.

My preference is to keep as is. I'm open to what others say though @mbosio85 - what say you?

File: docs/faq.md (hunk 3 above)
Comment (gwright99) [reply docs/faq.md:55]: I.e. push this to the Nextflow help docs? 

I'd need to understand more about what was discussed. My base position is that the Tower FAQ should provide guidance on issues that can express themselves if one uses the Tower platform (even if it is related to an underlying Nextflow problem), because the user is going to blame Tower first.

I don't mind compromising and having this question link back to the Nextflow site for a more fulsome explanation though.

File: docs/faq.md (hunk 5 above)
Comment (gwright99) [reply docs/faq.md:73]: I was under the impression this was a different file (I even complained about there being two different files with the same name). Was I mistaken and it's actually the same one?

File: docs/faq.md (hunk 6 above)
Comment (gwright99) [reply docs/faq.md:127]: Do we not think Tower Admins will consult the Help site? Moreover, I can imagine a situation where a Power User does initial problem triaging and is looking at docker logs and thinks "man, I really need some better logging - is this possible?" They then come to the site, see that it is possible, but realize they need to get the Admin involved to make the change for them.

File: docs/faq.md (hunk 2 above)
Comment (mbosio85) [reply docs/faq.md:50]: I think that it's more likely people will go and read `help.tower.nf` when looking for information about Tower and here I agree with @gwright99. Considering all is public there is no harm in showing it here as well (if it needs to be in the deployment docs).
The only caveat I have and it applies wherever this info will be published is: Does disclosing this information generates any harm to Seqera like showing potential weaknesses of our infrastructure? (I'm truly ignorant here so feel free to disregard my observation).

Ah, we can always ask whoever tracks it, to share with us the visits counts to help and deployment docs. I imagine that the help.tower.nf one has way more visits and, with that, potential to be useful.

File: docs/faq.md (hunk 7 above)
Comment (mbosio85) [reply docs/faq.md:211]: If the behavior differs we should say it, in my opinion.
Can we also say what is the behavior for users in tower.nf? (I believe it's the most recent one)

File: docs/faq.md (hunk 2 above)
Comment (abhi18av) [reply docs/faq.md:50]: Yes, the information is public - however we haven't (yet) mentioned `MICRONAUT_ENVIRONMENTS` etc in the help/Usage docs at all.

I'm happy to roll with the what has been suggested overall, single documentation is _much_ easier to maintain however when this idea was proposed in the past the consensus was to address a different set of users using these docs _separately_, the usage docs is mostly meant for people who treat Tower as a platform they use to run pipelines (a majority of them being what we call "Launchpad users") who don't have to take care of these underlying details.

This is also in the line (afaik) the SaaS plans where the platform details don't matter - think about Github and Github Enterprise. We barely see any reference of Github enterprise (or installation details) in the Github docs (https://docs.github.com/en).

Therefore, if the overall plan is to merge these docs (or have overlapping content) then I'd say we start sooner rather than later. But if we continue with the distinction with Usage/Installation docs, in that case we'd have to be pesky (😅 ) about maintaining the difference.

Thoughts?

Comment (abhi18av) [reply docs/faq.md:50]: Pinging the resident wizard 🪄  @evanfloden for the overall vision here.

TLDR; summary below

- Users often end up referring to both Usage and Deployment docs
- There are some documented features which need to be enabled via deployment configs of Tower
- Jumping between multiple doc sources is a bit annoying for anyone trying to "get stuff done"

File: docs/faq.md (hunk 5 above)
Comment (abhi18av) [reply docs/faq.md:73]: Na, you're spot on. These are different files but with the potential to confuse users.

File: docs/faq.md (hunk 7 above)
Comment (gwright99) [reply docs/faq.md:211]: Agree with Abhinav that 'Enterprise' wasn't required since this applies both to Enterprise and Cloud. Commit pushed with the change.

**Comment by abhi18av** [comment]
After a discussion with Graham today, we both agree that for the time being delivering faster information is of higher value than keeping it in the right docs and therefore we've decided to merge and publish this PR in its current state.

However, I'd like to add here that the content of the FAQ can be considered a bit volatile since we'd continue making changes there.
